File: 1221. Split a String in Balanced Strings/1221. Split a String in Balanced Strings.js
/**
 * @param {string} s
 * @return {number}
 */
var balancedStringSplit = function(s) {
    var n=s.length
    var max=0
    var balance=0
    for(let i=0; i<n; i++){
        if(s[i]==="R"){
            balance++
        }
        else if(s[i]==="L"){
            balance--
        }
        if(balance==0){
            max++
        }
    }
    return max
    
};

File: 1221. Split a String in Balanced Strings/1221. Split a String in Balanced Strings.py
class Solution:
    def balancedStringSplit(self, s: str) -> int:
        n=len(s)
        max=0
        balance=0
        for i in range(n):
            if s[i]=="R":
                balance+=1
            elif s[i]=="L":
                balance-=1
            if balance==0:
                max+=1
        return max
            
        
